polygon-validator-signer-balance-bot/src/agent.spec.ts

```diff
@@ -0,0 +1,161 @@
+import { FindingType, FindingSeverity, Finding, getJsonRpcUrl, ethers, HandleBlock, Initialize } from "forta-agent";
+import { provideHandleBlock, provideInitialize } from "./agent";
+import { TestBlockEvent } from "forta-agent-tools/lib/test/index";
+import { createAddress } from "forta-agent-tools/lib/utils";
+import { Flag, MINIMUM_THRESHOLD_TO_ETHER } from "./constants";
+
+describe("POLYGON-VALIDATOR-SIGNER BOT TEST SUITE", () => {
+  let handleBlock: HandleBlock;
+  let mockEthersProvider: any;
+  let initialize: Initialize;
+  let flag: Flag;
+  let mockUser: string;
+
+  beforeEach(async () => {
+    mockEthersProvider = {
+      getBalance: jest.fn(),
+    };
+    flag = { wasOverThresholdAlert: false };
+
+    mockUser = createAddress("0x01");
+    initialize = provideInitialize(mockEthersProvider as any, flag, mockUser);
+    handleBlock = provideHandleBlock(mockEthersProvider as any, flag, mockUser);
+  });
+
+  it("sets the flag to true when the balance is greater than the minimum threshold", async () => {
+    const accountBalance = ethers.utils.parseEther("10.0");
+    mockEthersProvider.getBalance = jest.fn(() => Promise.resolve(accountBalance));
+    await initialize();
+    expect(flag.wasOverThresholdAlert).toBe(true);
+    expect(mockEthersProvider.getBalance).toHaveBeenCalledWith(mockUser);
+  });
+
+  it("sets the flag to false when the balance is less than the minimum threshold", async () => {
+    const accountBalance = ethers.utils.parseEther("0.1");
+    mockEthersProvider.getBalance = jest.fn(() => Promise.resolve(accountBalance));
+    await initialize();
+    expect(flag.wasOverThresholdAlert).toBe(false);
+    expect(mockEthersProvider.getBalance).toHaveBeenCalledWith(mockUser);
+  });
+
+  it("throws an error when the provider fails to get the account balance", async () => {
+    mockEthersProvider.getBalance = jest.fn(() => Promise.reject("failed to fetch balance"));
+    try {
+      await initialize();
+    } catch (error) {
+      expect(error).toEqual("failed to fetch balance");
+      expect(mockEthersProvider.getBalance).toHaveBeenCalledWith(mockUser);
+    }
+  });
+
+  it("should return an array of findings when the account balance is below the threshold and an over threshold alert has been sent", async () => {
+    jest.spyOn(mockEthersProvider, "getBalance").mockResolvedValue(ethers.BigNumber.from("5000000000"));
+    flag.wasOverThresholdAlert = true;
+    const blockEvent = new TestBlockEvent().setNumber(10);
+    const findings = await handleBlock(blockEvent);
+    const balance = (await mockEthersProvider.getBalance(mockUser)).toString();
```

Comment:
```suggestion
    const balanceString = balance.toString();
```
Would recommend to use the mock return value like this to then assign it to another variable. Since you are setting the value of `balance` yourself, you can just use that.

If do change this to be `balanceString`, that would need to be changed elsewhere.